src/Interpreters/loadMetadata.cpp

```diff
@@ -221,13 +221,26 @@ void loadMetadata(ContextMutablePtr context, const String & default_database_nam
         databases.emplace(default_database_name, std::filesystem::path(path) / escapeForFileName(default_database_name));
     }
 
+    bool has_ordinary_databases = false;
     TablesLoader::Databases loaded_databases;
     for (const auto & [name, db_path] : databases)
     {
         loadDatabase(context, name, db_path, has_force_restore_data_flag);
-        loaded_databases.insert({name, DatabaseCatalog::instance().getDatabase(name)});
+
+        auto database = DatabaseCatalog::instance().getDatabase(name);
+        loaded_databases.insert({name, database});
+        
+
+        if (database->getEngineName() == "Ordinary")
+            has_ordinary_databases = true;
+        
+        loaded_databases.insert({name,database});
     }
 
+    if (has_ordinary_databases)
+        context->addWarningMessage("Server has databases with Ordinary engine, which is not optimal. \n " 
+            "To convert this database to a new Atomic engine, please create a flag named `convert_ordinary_to_atomic` in flags directory");
```

Comment:
We can move this line to https://github.com/ClickHouse/ClickHouse/blob/c4fb0a0cd520faeb61ac0cdb9b834062a558b1b4/src/Databases/DatabaseFactory.cpp#L166-L172 or to https://github.com/ClickHouse/ClickHouse/blob/8013cb1f784f6324b3c7b227499751dc7e666009/src/Databases/DatabaseOrdinary.cpp#L81-L84 so we will be able to remove other lines added in this PR, so it will be just one-line change

Reply:
With recent changes (when we include the name of the first database which has Ordinary) I think better to keep the code as is.

Reply:
Ok I moved the code to the Context.